assimilation_code/modules/assimilation/assim_tools_mod.pf.f90

```diff
@@ -1014,6 +1253,10 @@ subroutine filter_assim(ens_handle, obs_ens_handle, obs_seq, keys,           &
       endif
       whichvert_obs_in_localization_coord = nint(whichvert_real)
 
+      if (is_doing_vertical_conversion) then
+         ! use converted vertical coordinate value and type from owner
+         call set_vertical(base_obs_loc, vertvalue_obs_in_localization_coord, whichvert_obs_in_localization_coord)
+      endif
```

Comment:
This is inside the SEQUENTIAL_OBS do loop - I'm not sure it should be.
Should it be outside since looking at assm_tools_mod.f90 :
```
   !> all tasks must set the converted vertical values into the 'base' version of this loc
   !> because that's what we pass into the get_close_xxx() routines below.
```

Reply:
it does need to be inside the SEQUENTIAL_OBS do loop - it was before.  but it looks to me like these lines have moved to inside the 'not owner of this obs' section, instead of below the endif at line 1260 where the 'all tasks do this' section starts. (see comment at line 1263).  it looks like it should move down to after line 1270.

Reply:
yeah I meant in the 'not owner of this obs' section not SEQUENTIAL_OBS.  I'll move the lines to 1270. 
I wonder if the `! SEEMS TO BE NEEDED FOR VERTICAL LOCALIZATION IN BGRID` comment is related to not all the tasks setting the vertical values.

Reply:
the option to do all the vertical conversions up front for either obs, state or both before going into the sequential loop, vs just calling get_close_xx() and having that code convert locations "on demand" makes the flow a bit confusing. 

that option is useful for performance if the obs are only going to impact a small part of the state, or if a lot of the obs aren't going to be assimilated because of namelist choices or failed forward operators.

here, i think set_vertical() only needs to be done if the owning task has already converted the vertical, which is usually true for obs so there's no extra if() test added.  but it does make it look like obs are always already converted which isn't true depending on the namelist settings.  the get_close_xx() routines still need to be able to convert and update the locations if the vertical isn't already in the right coordinates.

in retrospect it would have been simpler to require all the locations to be converted at the start and then the get_close routines wouldn't have to have optional code for doing conversions on demand.  not sure the possible better performance for some unusual cases is worth the complexity.

Reply:
moved the `set vertical` to match assim_tools_mod.f90